com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -63,17 +65,35 @@ private void deleteService(CloudControllerClient client, String serviceName) {
 
     private void attemptToDeleteService(CloudControllerClient client, CloudServiceInstance serviceInstance, String serviceName) {
         List<CloudServiceBinding> bindings = serviceInstance.getBindings();
+        List<CloudApplication> applications = client.getApplications();

```

Comment:
This will return all applications in the space, not only the application that are bound to the service. It will end up in error in unbindServiceFromApplication().
You can check CreateOrUpdateServicesStep::getBoundAppNames which has similar logic.